### ISSUE TITLE
should I use with torch.no_grad() mode if my loss function needs to compute the gradients

### ISSUE THREAD

**Issue by LeZhengThu** [pytorch/pytorch#69217]
Hi, I know that using with torch.no_grad() when validating the model can save lots of time and memory. However, since part of my loss function needs to compute the norm of Jacobian, which is computed using torch.autograd.grad, so when I use with torch.no_grad(), there's always an error in validation. What should I do in this case?

**Comment by H-Huang** [comment]
When you use `torch.no_grad()` the `.grad` of the tensor is not populated. If you need this gradient value in your computation then you should not use `torch.no_grad()`
